cloud/pkg/common/monitor/monitor.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitor
+
+import (
+	"context"
+	"net/http"
+	"net/http/pprof"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/klog/v2"
+
+	beehivecontext "github.com/kubeedge/beehive/pkg/core/context"
+	config "github.com/kubeedge/kubeedge/pkg/apis/componentconfig/cloudcore/v1alpha1"
+)
+
+const (
+	metricNamespace = "KubeEdge"
+
+	// CloudHubSubsystem - subsystem name used by CloudHub
+	CloudHubSubsystem = "CloudHub"
+)
+
+var (
+	ConnectedNodes = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: CloudHubSubsystem,
+			Name:      "connected_nodes",
+			Help:      "Number of nodes that connected to the cloudHub instance",
+		},
+	)
+)
+
+var registerOnce sync.Once
+
+// registerMetrics register all metrics.
+func registerMetrics() {
+	registerOnce.Do(func() {
+		prometheus.MustRegister(
+			ConnectedNodes,
+		)
+	})
+}
+
+func InstallHandlerForPProf(mux *http.ServeMux) {
+	mux.HandleFunc("/debug/pprof/", pprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
```

Comment:
do we need /debug/pprof/heap ?

Reply:
It already exits in the `/debug/pprof/`

Reply:
It works if we don't add sth like "mux.HandleFunc("/debug/pprof/heap", pprof.Heap)"?

Reply:
yes, 

```
root@kubeedge-dev-linux-0004:~/gowork/src/github.com/kubeedge/kubeedge# go tool pprof http://127.0.0.1:9091/debug/pprof/heap
Fetching profile over HTTP from http://127.0.0.1:9091/debug/pprof/heap
Saved profile in /root/pprof/pprof.cloudcore.alloc_objects.alloc_space.inuse_objects.inuse_space.013.pb.gz
File: cloudcore
Build ID: 6aa303827980d57ac12a616d08d9cb9d746d4a73
Type: inuse_space
Time: Nov 9, 2022 at 3:10pm (CST)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top
Showing nodes accounting for 9119.46kB, 74.80% of 12192.26kB total
Showing top 10 nodes out of 105
      flat  flat%   sum%        cum   cum%
 1815.09kB 14.89% 14.89%  1815.09kB 14.89%  github.com/kubeedge/kubeedge/cloud/pkg/edgecontroller/controller.NewUpstreamController
 1033.55kB  8.48% 23.36%  1033.55kB  8.48%  regexp.onePassCopy
    1027kB  8.42% 31.79%  1539.01kB 12.62%  k8s.io/apimachinery/pkg/runtime.(*Scheme).AddKnownTypeWithName
 1025.94kB  8.41% 40.20%  1025.94kB  8.41%  regexp/syntax.(*compiler).inst (inline)
    1025kB  8.41% 48.61%     1025kB  8.41%  runtime.allocm
 1024.41kB  8.40% 57.01%  1024.41kB  8.40%  runtime.malg
  605.03kB  4.96% 61.97%   605.03kB  4.96%  github.com/kubeedge/beehive/pkg/core/channel.(*Context).newChannel
  536.37kB  4.40% 66.37%   536.37kB  4.40%  k8s.io/apimachinery/pkg/conversion.ConversionFuncs.AddUntyped
  514.38kB  4.22% 70.59%  1026.94kB  8.42%  k8s.io/api/core/v1.(*ConfigMapList).Unmarshal
  512.69kB  4.21% 74.80%  1025.31kB  8.41%  github.com/emicklei/go-restful.(*WebService).Route

```

Reply:
OK, thanks your reply